overlord/devicestate/handlers.go

```diff
@@ -108,7 +108,7 @@ func (m *DeviceManager) doMarkSeeded(t *state.Task, _ *tomb.Tomb) error {
 	if deviceCtx.HasModeenv() && deviceCtx.RunMode() {
 		// unset recovery_system because that is only needed during install mode
 		m.modeEnv.RecoverySystem = ""
-		err := m.modeEnv.Write("")
+		err := m.modeEnv.Write()
```

Comment:
here modeEnv though can be read or not. OTOH I start to dislike, be a bit worried about the way we keep a modeenv around like this. It will get out of sync on various boot package ops.

Reply:
oh it definitely already does, for example see the misadventures that led to adding `startOverlord` here: https://github.com/snapcore/snapd/blob/1d5040b8fd8c18c89240a1dacbb76373d3fa64c5/overlord/devicestate/firstboot_test.go#L107-L125

Reply:
Perhaps we should have another consistency check here with d.modeEvn.Unset()?